### ISSUE TITLE
Invocation context seems not to be properly passed when invoking traced functions

### ISSUE THREAD

**Issue by labbati** [DataDog/dd-trace-php#224]
Tested in the php 7.2 test runner. This is probably a different version of #149 

Invocation context seems not to be properly passed when invoking traced functions.

```
class A
{
    private function private_method()
    {
    }

    public function public_method()
    {
        spl_autoload_register(__CLASS__.'::private_method');
    }
}

dd_trace('spl_autoload_register', function () {
    return call_user_func_array('spl_autoload_register', func_get_args());
});

$a = new A();
$a->public_method();

$n = new \Not\Exist();
```

Results in 
```
Fatal error: Uncaught LogicException: Function 'A::private_method' not callable (cannot access private method A::private_method()) in /home/circleci/app/playground.php on line 16

LogicException: Function 'A::private_method' not callable (cannot access private method A::private_method()) in /home/circleci/app/playground.php on line 16

Call Stack:
    0.0010     393032   1. {main}() /home/circleci/app/playground.php:0
    0.0010     393752   2. A->public_method() /home/circleci/app/playground.php:20
    0.0010     394072   3. {closure:/home/circleci/app/playground.php:15-17}() /home/circleci/app/playground.php:11
    0.0011     394072   4. spl_autoload_register() /home/circleci/app/playground.php:16
```
While the expected output is
```
Fatal error: Uncaught Error: Class 'Not\Exist' not found in /home/circleci/app/playground.php on line 22

Error: Class 'Not\Exist' not found in /home/circleci/app/playground.php on line 22

Call Stack:
    0.0009     393032   1. {main}() /home/circleci/app/playground.php:0
```

**Comment by SammyK** [comment]
Just making a note that this issue will be resolved with #284 & #303. :)

**Comment by SammyK** [comment]
This should be fixed with [release `0.21.0`](https://github.com/DataDog/dd-trace-php/releases). :)
